Add merge_response templating (#34050)
home-assistant/home-assistant.github.io: source/_docs/configuration/templating.markdown

- Filter `value | regex_findall(find='', ignorecase=False)` will find all regex matches of the find expression in `value` and return the array of matches.
- Filter `value | regex_findall_index(find='', index=0, ignorecase=False)` will do the same as `regex_findall` and return the match at index.

## Merge action responses

Using action responses we can collect information from various entities at the same time.
Using the `merge_response` template we can merge several responses into one list.

| Variable       | Description                                     |
| -------------- | ----------------------------------              |
| `value`        | The incoming value (must be an action response). |

> `entity_id` key is appended to each dictionary within the template output list as a reference of origin. If the input dictionary already contains an `entity_id` key the template will fail.

Reviewer: ```suggestion
The `entity_id` key is appended to each dictionary within the template output list as a reference of origin. If the input dictionary already contains an `entity_id` key, the template will fail.
```

> `value_key` key is appended to each dictionary within the template output list as a reference of origin if the original service call was providing a list of dictionaries, as example `calendar.get_events` or `weather.get_forecasts`.

Reviewer: ```suggestion
The `value_key` key is appended to each dictionary within the template output list as a reference of origin if the original service call was providing a list of dictionaries, for example, `calendar.get_events` or `weather.get_forecasts`.
```

> Examples of these two keys can can be seen in [example merge calendar action response](#example-merge-calendar-action-response) template output.

Reviewer: ```suggestion
Examples of these two keys can be seen in [example merge calendar action response](#example-merge-calendar-action-response) template output.
```

### Example

```yaml
{% raw %}

{% set combined_forecast = merge_response(response) %}
{{ combined_forecast[0].precipitation | float(0) | round(1) }}

{% endraw %}
```

### Example how to sort

> Sorting the dictionaries within the list based on a specific key can be done directly by using the jinja's `sort` filter.

Reviewer: ```suggestion
Sorting the dictionaries within the list based on a specific key can be done directly by using Jinja's `sort` filter.
```

- Filter `value | regex_findall(find='', ignorecase=False)` will find all regex matches of the find expression in `value` and return the array of matches.
- Filter `value | regex_findall_index(find='', index=0, ignorecase=False)` will do the same as `regex_findall` and return the match at index.

## Merge action responses

Using action responses we can collect information from various entities at the same time.
Using the `merge_response` template we can merge several responses into one list.

| Variable     | Description                                            |
| ------------ | ----------------------------------                     |
| `value`      | The incoming value (action response).                  |
| `sort_by`    | Sort a response dictionary by this key.                |
| `single_key` | Return a list with information from a single key only. |

{% note %}

> `sort_by` and `single_key` has only use if the action call is returning dictionaries

Author: ```suggestion
`sort_by` and `single_key` have only use if the action call is returning dictionaries, for example `weather` and `calendar` action responses.
```

- Filter `value | regex_findall(find='', ignorecase=False)` will find all regex matches of the find expression in `value` and return the array of matches.
- Filter `value | regex_findall_index(find='', index=0, ignorecase=False)` will do the same as `regex_findall` and return the match at index.

## Merge action responses

Using action responses we can collect information from various entities at the same time.

> Using the `merge_response` template we can merge several responses into one list.

Reviewer 2: We should explain the `entity_id` will be added to the template output, and that the function will fail if there's already a key `entity_id`.

Also, let's add an example for the case where the response is a scalar.
